### PR TITLE
[AMDGPU] Fix v_dot2_f16_f16/v_dot2_bf16_bf16 operands

### LINKED ISSUE #82038: [AMDGPU][AsmParser] 32-bit integer can't be used as PK2_BF16 operand
The assembler doesn't recognize cases such as `v_dot2_bf16_bf16 v2, s0, 0x3f803f80, v2` where `0x3f803f80` stands for `<bfloat 1.0, bloat 1.0>`.

### PR DESCRIPTION
**Pull request by rampitec**
src0 and src1 are packed f16/bf16, we are printing literals like 0x40002000, but we cannot parse it.

### CODE REVIEW

File: llvm/test/MC/Disassembler/AMDGPU/gfx11_dasm_vop3.txt [hunk 1]
```diff
@@ -1833,6 +1839,12 @@
 # GFX11: v_dot2_f16_f16 v255, -|0xfe0b|, -|vcc_hi|, null op_sel:[0,0,0,1] ; encoding: [0xff,0x43,0x66,0xd6,0xff,0xd6,0xf0,0x61,0x0b,0xfe,0x00,0x00]
 0xff,0x43,0x66,0xd6,0xff,0xd6,0xf0,0x61,0x0b,0xfe,0x00,0x00
 
+# GFX11: v_dot2_f16_f16 v2, v0, 0x20004000, v2 ; encoding: [0x02,0x00,0x66,0xd6,0x00,0xff,0x09,0x04,0x00,0x40,0x00,0x20]
+0x02,0x00,0x66,0xd6,0x00,0xff,0x09,0x04,0x00,0x40,0x00,0x20
+
+# GFX11: v_dot2_f16_f16 v2, 0x20004000, v0, v2 ; encoding: [0x02,0x00,0x66,0xd6,0xff,0x00,0x0a,0x04,0x00,0x40,0x00,0x20]
+0x02,0x00,0x66,0xd6,0xff,0x00,0x0a,0x04,0x00,0x40,0x00,0x20
```
Comment (shiltian) [inline llvm/test/MC/Disassembler/AMDGPU/gfx11_dasm_vop3.txt:1846]: How will it look like if `0x3f803f80` which represents `<bfloat 1.0, bfloat 1.0>`?

**Review by shiltian — COMMENTED** [review]
I suppose this can fix #82038?

**Comment by rampitec** [comment]
> I suppose this can fix #82038?

Yes

File: llvm/test/MC/Disassembler/AMDGPU/gfx11_dasm_vop3.txt (hunk 1 above)
Comment (rampitec) [reply llvm/test/MC/Disassembler/AMDGPU/gfx11_dasm_vop3.txt:1846]: Exactly like this: 0x3f803f80
```
v_dot2_f16_f16 v2, v0, 0x3f803f80, v2   ; encoding: [0x02,0x00,0x66,0xd6,0x00,0xff,0x09,0x04,0x80,0x3f,0x80,0x3f]
```

Comment (rampitec) [reply llvm/test/MC/Disassembler/AMDGPU/gfx11_dasm_vop3.txt:1846]: I.e. there is no difference if that is 0x3f803f80 or 0x20004000. This is not an inline constant that way or another.

Comment (shiltian) [reply llvm/test/MC/Disassembler/AMDGPU/gfx11_dasm_vop3.txt:1846]: 1.0 is inline literal right? does it also apply to packed operand?

Comment (rampitec) [reply llvm/test/MC/Disassembler/AMDGPU/gfx11_dasm_vop3.txt:1846]: 1.0 is inline, but <1.0, 1.0> is not. You could use another form with inline 1.0 and op_sel_hi, but that is another instruction.

Comment (rampitec) [reply llvm/test/MC/Disassembler/AMDGPU/gfx11_dasm_vop3.txt:1846]: And then `v_dot2_bf16_bf16 v2, v0, 0x3f80, v2` gives inline 1.0:
```
v_dot2_bf16_bf16 v2, v0, 1.0, v2        ; encoding: [0x02,0x00,0x67,0xd6,0x00,0xe5,0x09,0x04]
```

**Review by shiltian — APPROVED** [review]
LGTM. Thanks!

File: llvm/test/MC/AMDGPU/gfx11_asm_vop3.s [hunk 2]
```diff
@@ -2116,6 +2116,12 @@ v_dot2_bf16_bf16 v5, -src_scc, |vcc_lo|, -1 op_sel:[0,0,1,0]
 v_dot2_bf16_bf16 v255, -|0xfe0b|, -|vcc_hi|, null op_sel:[0,0,0,1]
 // GFX11: encoding: [0xff,0x43,0x67,0xd6,0xff,0xd6,0xf0,0x61,0x0b,0xfe,0x00,0x00]
 
+v_dot2_bf16_bf16 v2, v0, 0x20004000, v2
+// GFX11: v_dot2_bf16_bf16 v2, v0, 0x20004000, v2 ; encoding: [0x02,0x00,0x67,0xd6,0x00,0xff,0x09,0x04,0x00,0x40,0x00,0x20]
+
+v_dot2_bf16_bf16 v2, 0x20004000, v0, v2
```
Comment (arsenm) [inline llvm/test/MC/AMDGPU/gfx11_asm_vop3.s:2122]: do we need tests with op_sel + constant?

Comment (rampitec) [reply llvm/test/MC/AMDGPU/gfx11_asm_vop3.s:2122]: There is nothing in the MC which ties op_sel/op_sel_hi to a fact an operand is a literal, so the answer is probably no. But then I do not see any tests with op_sel_hi at all.

**Pull request merged**